### ISSUE TITLE
Training from scratch

### ISSUE THREAD

**Issue by mattgallivan** [Harmonai-org/sample-generator#10]
Hi, thanks for the excellent library! How would I train the model from scratch from a folder full of .wav files?

**Comment by twobob** [comment]
put them in a folder. that is pointed to by TRAINING_DIR

!python3 /content/sample-generator/train_uncond.py\
                                                          --name $NAME\
                                                          --training-dir $TRAINING_DIR\
                                                          --sample-size $SAMPLE_SIZE\
                                                          --accum-batches $ACCUM_BATCHES\
                                                          --sample-rate $SAMPLE_RATE\
                                                          --batch-size $BATCH_SIZE\
                                                          --demo-every $DEMO_EVERY\
                                                          --demo-steps 125\
                                                          --num-demos 4\
                                                          --checkpoint-every $CHECKPOINT_EVERY\
                                                          --num-workers 2\
                                                          --num-gpus 1\
                                                          $random_crop_str\
                                                          --save-path $OUTPUT_DIR\

so elide  --ckpt-path $CKPT_PATH\

and it will train from scratch

**Comment by ctralie** [comment]
What effective batch size would you recommend when training from scratch?

**Comment by twobob** [comment]
> What effective batch size would you recommend when training from scratch?

8 accumulated and a batch of 2 would be safe IIRC. Clearly think big but those are "nice" numbers

**Comment by ctralie** [comment]
Ah okay that's interesting, thanks.  Due to memory constraints on my rig, I've been doing a batch size of 4 and accumulated 8 to reach the magic number 32, which has been working well.  But it's good to know I might be able to get away with a smaller effective batch size.  I noticed that no batch normalization is happening; only group normalization.  So I suppose the batch size will have less of an impact than it would otherwise?

**Comment by twobob** [comment]
quite possibly :)

Maybe switch over to stable-audio-tools

https://github.com/Stability-AI/stable-audio-tools
https://huggingface.co/stabilityai/stable-audio-open-1.0

as you can see from the notes and the older configs that is the next generation of this.
https://github.com/Stability-AI/stable-audio-tools/tree/main/stable_audio_tools/configs/model_configs/dance_diffusion


here is another take on that with a nod to creating the 2_0 model with it's much longer times.
https://github.com/yukara-ikemiya/friendly-stable-audio-tools
Just some random thoughts, hope they help

**Comment by ctralie** [comment]
Thanks for the links!
Yes, this new version seems very cool.  But actually, for a research project I'm working on, it's great to have a very simple diffusion model at the raw audio sample level, rather than at the latent level, and which also doesn't have a language model with it.  So "dance diffusion" is perfect.
